src/v2/tags.rs

```diff
@@ -20,53 +20,66 @@ struct TagsChunk {
 impl Client {
     /// List existing tags for an image.
     pub fn get_tags(&self, name: &str, paginate: Option<u32>) -> StreamTags {
-        let dclient = self.clone();
+        let client = self.clone();
         let base_url = format!("{}/v2/{}/tags/list", self.base_url, name);
 
-        let fres = futures::stream::unfold(Some(String::new()), move |link| {
+        let fres = futures::stream::unfold(Some(String::new()), move |last| {
+            let client = client.clone();
+
             // Stream ends when response has no `Link` header.
-            let last = match link {
+            let last = match last {
                 None => return None,
                 Some(ref s) if s == "" => None,
                 s => s,
             };
 
-            let full_url = match (paginate, last) {
+            let url_paginated = match (paginate, last) {
                 (Some(p), None) => format!("{}?n={}", base_url, p),
                 (None, Some(l)) => format!("{}?next_page={}", base_url, l),
                 (Some(p), Some(l)) => format!("{}?n={}&next_page={}", base_url, p, l),
                 _ => base_url.to_string(),
             };
-            let client = dclient.clone();
-            let url = hyper::Uri::from_str(&full_url);
-
-            let freq = futures::future::result(url)
-                .from_err()
-                .and_then(move |url| {
-                    trace!("GET {:?}", &url);
-                    let req = client.new_request(hyper::Method::GET, url);
-                    futures::future::result(req)
-                        .and_then(move |req| client.hclient.request(req).from_err())
+
+            let freq = futures::future::result(Url::parse(&url_paginated))
+                .map_err(|e| Error::from(format!("{}", e)))
+                .inspect(|url| trace!("GET {}", url))
+                .map(move |url| {
+                    // receive the next page of tags
+
+                    client
+                        .build_reqwest(reqwest::async::Client::new().get(url.clone()))
+                        .header(header::ACCEPT, "application/json")
+                        .send()
+                        .map_err(|e| Error::from(format!("{}", e)))
                 })
+                .flatten()
                 .and_then(|resp| {
+                    // ensure the status is OK
```

Comment:
I think `reqwest` has a `error_for_status` which should implement this same logic.